### ISSUE TITLE
How do i make looping script and then make it stop somehow?

### ISSUE THREAD

**Issue by picarica** [autokey/autokey#417]
## Classification:

Help

## Reproducibility:

Always

## Version

AutoKey version: 0.90.4

Used GUI (Gtk, Qt, or both): Gtk

If the problem is known to be present in more than one version, please list all of those.

Installed via: .ebuild

Linux Distribution: Gentoo

## Summary
so i want to make script loop OR copy & paste code that it goes forever but then i need to somehow force close the script, i have set window filter but even when active window changes the already run script goes

so i have simple code 

```
keyboard.send_key("<tab>")
time.sleep(0.5)
keyboard.send_key("<enter>")
time.sleep(2)
keyboard.send_keys("<ctrl>+w")
time.sleep(2.5)
```
i want this to repeat for as many tabs i have open which is unpredictable
how do i do  this ?

**Comment by ineuw** [comment]
You will need to call on some Python libraries.
import os
import sys
are always a safe bet.
Figure out or capture the number of open tabs from the browser.
Use this number to iterate in a For loop around your code.

Unfortunately, I am not a coder so you have to read up on how to implement ths.

**Comment by picarica** [comment]
> You will need to call on some Python libraries.
> import os
> import sys
> are always a safe bet.
> Figure out or capture the number of open tabs from the browser.
> Use this number to iterate in a For loop around your code.
> 
> Unfortunately, I am not a coder so you have to read up on how to implement ths.

intresting idea ill try to look up on that

**Comment by josephj11** [comment]
When AutoKey detects that you have pressed a hotkey or typed an abbreviation which you have defined in one or more phrases or scripts, it checks to see if the window filter for each such phrase or script matches the current window title and class. If either one matches the filter, the phrase or script is executed. This is a one-time action. It doesn't do anything further until you trigger it again with another hotkey or abbreviation.

I took a quick look and didn't see an easy way to discover tabs/tab counts.
I'm no Python expert, so I often look for workarounds with things I do understand.

In your current example, you could look at the current  window title or class with `window.get_active_title()` or `window.get_active_class()`.

Assuming that `ctrl+w` is closing tabs, sooner or later, they'll all be closed forcing the focus to shift to another window which, hopefully, won't have the same title or class, so you'll know to quit. Of course, many applications such as browsers allow you to open multiple windows, each with multiple tabs in them, so you'll have to either avoid that situation or figure out how to handle it. (I think I know how to do that if you need to.)

We appreciate and support usage questions, but they should be made on our [support list](https://groups.google.com/forum/#!forum/autokey-users) or, for more advanced questions, on our [Gitter page](https://gitter.im/autokey/autokey).

**Comment by ineuw** [comment]
Are you trying to identify tabs of a browser and if so, which browser?

**Comment by josephj11** [comment]
@ineuw Yeah. If that's what the OP wants to do, I think I saw a couple of relevant links for Firefox and another browser.

**Comment by ineuw** [comment]
I can only think of Mozillazine.
